custom-tests.yaml

```diff
@@ -214,8 +214,13 @@ api:
       decodeAudioData.promise_syntax: >-
         <%api.BaseAudioContext:ctx%>
         return ctx.decodeAudioData.length === 1;
-  # BatteryManager: # This promise seems to remain pending when running all tests
-  #   __base: var promise = navigator.getBattery();
+  BatteryManager:
+    __base: >-
+      if (!('getBattery' in navigator)) {
+        return false;
+      };
+      var promise = navigator.getBattery();
+      promise.then(function() {});
```

Comment:
Why is this `promise.then(function() {})` needed? It doesn't silence any rejected promise and seems like a no-op.

Reply:
It's an oddball statement, but there seems to be a bug in early Firefox versions where the promises from the navigator will daisy-chain `.then` statements even when using separate calls to `getBattery()` or `getUserMedia()`.  This seems to resolve the issue, though I couldn't say how or why.